### ISSUE TITLE
Variable may not be declared after bundling

### ISSUE THREAD

**Issue by MikeWillisFC** [webpack/webpack#7901]
# Bug report

**What is the current behavior?**
After webpack runs, a variable may wind up being declared in a spot that never executes

**If the current behavior is a bug, please provide the steps to reproduce.**
Run the following js through webpack:
```
if ( window.SomeFlag ) {
   for( var i = 0; i <= 5; i++ ) {
      console.log("hello");
   }
}

if ( window.SomeOtherFlag ) {
   for( var i = 0; i <= 5; i++ ) {
      console.log("hello again");
   }
}
```

The relevant portion of the resulting dist js comes out to the following on my machine:
```
function(e, t, n) {
        "use strict";
        if (window.SomeFlag)
            for (var r = 0; r <= 5; r++) console.log("hello");
        if (window.SomeOtherFlag)
            for (r = 0; r <= 5; r++) console.log("hello again")
    }
``` 

Now if window.SomeFlag is false and window.SomeOtherFlag is true, we get an error when the second loop attempts to use variable r (formerly i before bundling)

**What is the expected behavior?**
Perhaps move the declaration to the top of the parent function. In my case I fixed it by changing var to let on the source script. Then the resulting bundled script declares different variables in each loop.

**Other relevant information:**
webpack version:  4.16.5
Node.js version: 6.11.4
Operating System: Windows 10
Additional tools:

**Comment by montogeek** [comment]
This is an Uglify issue.

**Comment by MikeWillisFC** [comment]
The resulting bundled code for me showed as I posted above, without `var r` hoisted to function scope, which creates a "variable is not defined in strict mode" error.

Your input code is not the same as mine. You have `if(false)` - in this situation webpack seems to be aware that the `var x` declaration will never run, and it moves the declaration elsewhere. My code is using flags that webpack cannot know the value of, and in this situation it leaves the declaration where it was first listed. Try bundling the example code I listed and see if your output is the same?

**Comment by sokra** [comment]
I don't see the issue. Which browser are you using?

**Comment by MikeWillisFC** [comment]
Google Chrome.

Looks like you're right - I ran my sample code through a basic test page and sure enough I got no error. However, I did get the "variable undefined" error in my app. It was caught by Sentry and reported to me, pointing exactly back to the spot where the variable was used but not declared. I'm not sure why it generated an error in app but not on a test page. There must be something else going on that I didn't realize.

Sorry for the false report. If I figure out specifically how to reproduce I'll update here.
